File: src/main/java/org/bk/graphql/web/dto/BookDto.java
package org.bk.graphql.web.dto;

import org.bk.graphql.domain.Book;

public record BookDto(String id, String name, int pageCount, int version) {
    public static BookDto map(Book book) {
        return new BookDto(book.id().id(), book.name(), book.pageCount(), book.version());
    }
}

File: src/main/kotlin/org/bk/graphql/web/dto/BookDto.kt
package org.bk.graphql.web.dto

import org.bk.graphql.domain.Book

data class BookDto(
    val id: String,
    val name: String,
    val pageCount: Int,
    val version: Int = 0
) {
    companion object {
        fun map(book: Book): BookDto = BookDto(book.id.id, book.name, pageCount = book.pageCount, version = book.version)
    }
}